cmd/recoveretcdcluster/recoveretcdcluster.go

```diff
@@ -0,0 +1,53 @@
+package main
+
+import (
+	"context"
+	"os"
+
+	"github.com/sirupsen/logrus"
+	kerrors "k8s.io/apimachinery/pkg/util/errors"
+
+	"github.com/openshift/openshift-azure/pkg/api"
+	"github.com/openshift/openshift-azure/pkg/fakerp"
+	"github.com/openshift/openshift-azure/pkg/plugin"
+	"github.com/openshift/openshift-azure/pkg/util/managedcluster"
+)
+
+func main() {
+	logrus.SetLevel(logrus.DebugLevel)
+	logrus.SetFormatter(&logrus.TextFormatter{FullTimestamp: true})
+	log := logrus.NewEntry(logrus.StandardLogger())
+	if len(os.Args) != 2 {
+		log.Fatal("Usage recoveretcdcluster <blobname>")
+	}
+	blobName := os.Args[1]
+	conf, err := fakerp.NewConfig(log, true)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	ctx := context.Background()
+	ctx = context.WithValue(ctx, api.ContextKeyClientID, conf.ClientID)
+	ctx = context.WithValue(ctx, api.ContextKeyClientSecret, conf.ClientSecret)
+	ctx = context.WithValue(ctx, api.ContextKeyTenantID, conf.TenantID)
+
+	config, err := fakerp.GetPluginConfig()
```

Comment:
fakerp was never meant to be used in production code and recoveretcdcluster is production code. We should probably move the fakerp functions used here in a different package and ensure they are properly unit-tested. Please open follow-up issue(s) accordingly

@jim-minter

Reply:
https://github.com/openshift/openshift-azure/pull/798 needs reviews